RecoBTag/Combined/src/HeavyIonCSVTagger.cc

```diff
@@ -91,17 +88,14 @@ float HeavyIonCSVTagger::discriminator(const TagInfoHelper &tagInfo) const {
     //single value tagging var
     else {
       inputs[mva_var.name] = vars.get(mva_var.id, mva_var.default_value);
-      if (!vtxChecked) {
+      if (noTrack) {
         if (mva_var.name == "TagVarCSV_vertexMass") {
-          vtxMassVal = inputs[mva_var.name];
-          vtxChecked = true;
```

Comment:
At this point you can set `noTrack = false;` (or give it a different name if `noTrack` does not fit any more with the updated functionality) and get rid of these checks in the following cycles, as you did before with  `vtxChecked = true;`